> []{#_bookmark0 .anchor}**Polygenic Risk Score Analyses Workshop 2022**

![](media/image1.jpeg){width="2.94in" height="2.8874989063867016in"}

> **Day 3: Advanced Polygenic Risk Score Analyses**

# Day 3 Timetable {#day-3-timetable .unnumbered}

+------------+--------+-------------------------+---------------------+
| > []{#     | > **T  |                         | > **Presenter**     |
| _bookmark2 | itle** |                         |                     |
| > .ancho   |        |                         |                     |
| r}**Time** |        |                         |                     |
+============+========+=========================+=====================+
| > 9:00 -   | A      |                         | > \-                |
| > 9:15     | rrival |                         |                     |
+------------+--------+-------------------------+---------------------+
| > 9:15 -   | [Lec   | > Advanced PRS Analyses | > Dr Paul O'Reilly  |
| > 10:30    | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
+------------+--------+-------------------------+---------------------+
| 10:30 -    | >      |                         | > \-                |
| 11:00      |  Coﬀee |                         |                     |
|            | >      |                         |                     |
|            |  Break |                         |                     |
|            | > and  |                         |                     |
|            | > Q&A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 11:00 -    | > [Lec |                         | > Dr Paul O'Reilly  |
| 12:00      | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | > Path |                         |                     |
|            | way-sp |                         |                     |
|            | ecific |                         |                     |
|            | > PRS  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 12:00 -    | >      |                         | > \-                |
| 13:30      |  Lunch |                         |                     |
+------------+--------+-------------------------+---------------------+
| 13:30 -    | >      |                         | > Dr Conrad Iyegbe  |
| 14:30      | [Pract |                         | > & Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | > Ad   |                         |                     |
|            | vanced |                         |                     |
|            | > PRS  |                         |                     |
|            | > An   |                         |                     |
|            | alyses |                         |                     |
+------------+--------+-------------------------+---------------------+
| 14:30 -    | >      |                         | > \-                |
| 15:00      |  Coﬀee |                         |                     |
|            | >      |                         |                     |
|            |  Break |                         |                     |
|            | > and  |                         |                     |
|            | > Q&A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 15:00 -    | >      |                         | > Dr Conrad Iyegbe  |
| 16:00      | [Pract |                         | > & Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | > P    |                         |                     |
|            | athway |                         |                     |
|            | -based |                         |                     |
|            | > PRS  |                         |                     |
|            | > An   |                         |                     |
|            | alyses |                         |                     |
+------------+--------+-------------------------+---------------------+
| 16:00 -    | > [S   |                         | > Dr Segun Fatumo   |
| 17:00      | pecial |                         |                     |
|            | > Sem  |                         |                     |
|            | inar]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | > Pre  |                         |                     |
|            | cision |                         |                     |
|            | > Me   |                         |                     |
|            | dicine |                         |                     |
|            | > &    |                         |                     |
|            | >      |                         |                     |
|            | > \+   |                         |                     |
|            | > PRS  |                         |                     |
|            | > in   |                         |                     |
|            | >      |                         |                     |
|            | Africa |                         |                     |
+------------+--------+-------------------------+---------------------+

> **Contents**

[Day 3 Timetable](#day-3-timetable) 1

[D](#day-3-timetable)[ay 3 Timetable](#_bookmark2) . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . 1

1.  [Advanced Polygenic Risk Score
    Analyses](#advanced-polygenic-risk-score-analyses) 3

    1.  [Key Learning Outcomes](#key-learning-outcomes) . . . . . . . .
        . . . . . . . . . . . . . . . . 3

    2.  [Resources you will be using](#resources-you-will-be-using) . .
        . . . . . . . . . . . . . . . . . . . . 3

2.  [Advance Polygenic Score
    Analysis](#advance-polygenic-score-analysis) 4

    1.  [Key Learning Outcomes](#key-learning-outcomes-1) . . . . . . .
        . . . . . . . . . . . . . . . . . 4

    2.  [Resources](#resources) . . . . . . . . . . . . . . . . . . . .
        . . . . . . . . . . . . 4

    3.  [Over-fitting caused by model
        optimisation](#over-fitting-caused-by-model-optimisation) . . .
        . . . . . . . . . . . 6

    4.  [Distribution of PRS](#distribution-of-prs) . . . . . . . . . .
        . . . . . . . . . . . . . . . . . 8

    5.  [Gene Set Analysis](#gene-set-analysis) 10

        1.  [Molecular Signatures Database
            (MSigDB)](#molecular-signatures-database-msigdb) 10

        2.  [General Transfer Format
            file](#general-transfer-format-file) 11

        3.  [Browser Extensible Data
            (BED)](#browser-extensible-data-bed) 11

        4.  [Gene-Set Enrichment
            Analysis](#gene-set-enrichment-analysis) 12

    6.  [Gene-Set Based PRS Analysis](#gene-set-based-prs-analysis) 12

# Advanced Polygenic Risk Score Analyses

## Key Learning Outcomes

> After completing this practical, you should be able to:

1.  Perform basic [Polygenic Risk Score (PRS)](#_bookmark0) analyses
    using PRSice (Euesden et al. 2015; Choi & O'Reilly 2019)

2.  Interpret results generated from [PRS](#_bookmark0) analyses

3.  Customize visualization of results

## Resources you will be using

> To perform [PRS](#_bookmark0) analyses, summary statistics from
> [Genome Wide Association Stud-](#_bookmark0) [ies (GWAS)](#_bookmark0)
> are required. In this workshop, the following summary statistics are
> used:
>
> Phenotype Provider Description Download Link
>
> Height [GIANT
> Consortium](https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
> [GWAS](#_bookmark0) of height on
>
> 253,288 individuals (Wood et al. 2014)
>
> [Download](https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz)
> [Link](https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz)
>
> [Coronary](#_bookmark0) [artery disease](#_bookmark0)
> [(CAD)](#_bookmark0)
>
> [CARDIoGRAM plus](http://www.cardiogramplusc4d.org/) [C4D
> consortium](http://www.cardiogramplusc4d.org/)
>
> [GWAS](#_bookmark0) on 60,801 CAD
>
> cases and 123,504 con- trols (CAD Consortium 2015)
>
> [Download](http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip)
> [Link](http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip)
>
> ![](media/image2.jpeg){width="0.3229166666666667in"
> height="0.3229166666666667in"}

# Advance Polygenic Score Analysis

## Key Learning Outcomes

> After completing this practical, you should be able to:

1.  Know how to adjust for ascertainment bias in case-control
    [PRS](#_bookmark0) analysis

2.  Know how over-fitting aﬀects [PRS](#_bookmark0) results and how to
    handle it

3.  Understand distribution of [PRS](#_bookmark0)

4.  Understand diﬀerent file formats involved in gene-set analysis

5.  Understand diﬀerence between self-contained and competitive gene-set
    analy- ses

6.  Calculate pathway based [PRS](#_bookmark0)

## Resources

> In this practical, we will explore how to perform gene-set based
> [PRS](#_bookmark0) analyses. To perform this analysis, gene-set
> information and coordinates for the genic regions are required. These
> information can be obtained from the following database:
>
> Data Set Description Download Link
>
> Ensembl Human Genome GTF file
>
> A file containing the coordinates for genes in the human genome. Used
> by PRSice to map the SNPs onto genic regions.
>
> [here](ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens)
>
> MSigDB Gene Sets File containing the gene-set informa-
>
> tion. *Free registration required.*
>
> [Download](http://software.broadinstitute.org/gsea/msigdb/download_file.jsp?filePath=/resources/msigdb/6.1/h.all.v6.1.symbols.gmt)
> [here
> after](http://software.broadinstitute.org/gsea/msigdb/download_file.jsp?filePath=/resources/msigdb/6.1/h.all.v6.1.symbols.gmt)
> [registration](http://software.broadinstitute.org/gsea/msigdb/download_file.jsp?filePath=/resources/msigdb/6.1/h.all.v6.1.symbols.gmt)

2.  RESOURCES

> \(a\) Nagelkerke*R*^2^ (b) Lee adjusted *R*^2^

![](media/image3.jpeg){width="2.135in"
height="2.1590616797900264in"}![](media/image4.jpeg){width="2.135in"
height="2.1590616797900264in"}

> Figure 2.1: Performance of diﬀerent *R*^2^ when the study contains
> equal portion of cases and controls

![](media/image5.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}![](media/image2.jpeg){width="0.3229166666666667in"
height="0.3229166666666667in"}

> **Estimating** *R*^2^ **in case/control studies**
>
> Bias in *R*^2^ estimation caused by ascertained case/control samples
> can be adjusted using the equation proposed by Lee et al (2011), which
> requires the sample preva- lence (case/control ratio) and population
> prevalence as parameters. This function is implemented in PRSice and
> the adjustment can be performed by providing the population prevalence
> to the command \--prevalence.
>
> Residuals of logistic regression is not well defined, and in
> [PRS](#_bookmark0) analyses, Nagelkerke *R*^2^ is usually used to
> represent the model *R*^2^ (this is the default of PRSice). However,
> this *R*^2^ does not account for the diﬀerence between sample
> prevalence (i.e. case- control ratio) and population prevalence, which
> can lead to bias in the reported *R*^2^ (Fig.2.1a).
>
> Bias in *R*^2^ estimation caused by ascertained case/control samples
> can be adjusted using the equation proposed by Lee et al. 2012
> (Fig.2.1b), which requires the sample prevalence (case/control ratio)
> and population prevalence as parameters. This func- tion is
> implemented in PRSice and the adjustment can be performed by providing
> the population prevalence to the command \--prevalence.
>
> Now, account for the ascertainment of the case/control sample by
> including the population prevalence (let's assume e.g. 5% here) in the
> PRSice command to obtain
>
> the adjusted (Lee) *R*^2^:
>
> 1 Rscript ./Software/PRSice.R \\
>
> 2 \--dir . \\
>
> 3 \--prsice Software/PRSice_linux \\
>
> 4 \--base Base_Data/cad.add.txt \\
>
> 5 \--target Target_Data/TAR \\
>
> 6 \--snp markername \\
>
> 7 \--A1 eﬀect_allele \\
>
> 8 \--A2 noneﬀect_allele \\
>
> 9 \--chr chr \\
>
> 10 \--bp bp_hg19 \\
>
> 11 \--stat beta \\
>
> 12 \--beta \\
>
> 13 \--pvalue p_dgc \\
>
> 14 \--pheno Target_Data/TAR.cad \\
>
> 15 \--prevalence 0.05 \\
>
> 16 \--binary-target T \\
>
> 17 \--out Results/CAD.highres.LEER2
>
> Check the \*.summary file in the Results/ folder where you will find
> the usual (Nagelkerke) *R*^2^ and the adjusted (Lee) *R*^2^.

![](media/image6.jpeg){width="0.3541666666666667in"
height="0.3593744531933508in"}

## Over-fitting caused by model optimisation

> In PRS analyses, the shrinkage or tuning parameter is usually
> optimized across a wide range of parametric space (e.g. *P* -value
> threshold, proportion of causal SNPs). When both optimisation and
> association testing are performed on the target data, over-fitted
> results will be obtained. The accuracy and predictive power of
> over-fitted results are likely to diminished when replicated in an
> independent data set.
>
> A simple solution is to perform permutation to obtain an empirical *P*
> -value for the association model, which is implemented in PRSice.
> Briefly, permutation is performed as follows:

1)  Compute the *P* -value in your original data, denoted as *obs.p*, at
    the \"best\" threshold.

2)  []{#_bookmark10 .anchor}Then shuﬄe the phenotype and obtain the *P*
    -value of the \"best\" threshold for this null phenotype, denoted as
    *null.p*

3)  Repeat **[2)](#_bookmark10)** *N* times

4)  Calculate the empirical *P* -value as:

> (P(*obs.p \> null.p~i~*)+ 1)

*Pemp* =

> (*N* + 1)
>
> You will have to specify the number of permutation (*N* ) to perform
> by providing
>
> \--perm N as a parameter to PRSice.
>
> 1 Rscript ./Software/PRSice.R \\
>
> 2 \--prsice Software/PRSice_linux \\
>
> 3 \--base Base_Data/GIANT_Height.txt \\
>
> 4 \--target Target_Data/TAR \\
>
> 5 \--snp MarkerName \\
>
> 6 \--A1 Allele1 \\
>
> 7 \--A2 Allele2 \\
>
> 8 \--stat b \\
>
> 9 \--beta \\
>
> 10 \--pvalue p \\
>
> 11 \--pheno Target_Data/TAR.height \\
>
> 12 \--binary-target F \\
>
> 13 \--cov Target_Data/TAR.covariate \\
>
> 14 \--cov-col Sex \\
>
> 15 \--perm 1000 \\
>
> 16 \--out Results/Height.perm

![](media/image7.jpeg){width="0.3281244531933508in"
height="0.3281244531933508in"}![](media/image6.jpeg){width="0.3541666666666667in"
height="0.3593744531933508in"}

> The best way to avoid having results that are over-fit is to perform
> validation on an independent validation data set. We can perform
> validation of the previous height
>
> \+ covariate analysis with PRSice, using the independent VAL target
> sample as validation data and the \"best\" *P* -value threshold
> predicted in the VAL samples:
>
> 1 Rscript ./Software/PRSice.R \\
>
> 2 \--prsice Software/PRSice_linux \\
>
> 3 \--base Base_Data/GIANT_Height.txt \\
>
> 4 \--target Target_Data/VAL \\
>
> 5 \--snp MarkerName \\
>
> 6 \--A1 Allele1 \\
>
> 7 \--A2 Allele2 \\
>
> 8 \--stat b \\
>
> 9 \--beta \\
>
> 10 \--pvalue p \\
>
> 11 \--pheno Target_Data/VAL.height \\
>
> 12 \--binary-target F \\
>
> 13 \--no-full \\
>
> 14 \--bar-levels 0.0680001 \\
>
> 15 \--fastscore \\
>
> 16 \--cov Target_Data/VAL.covariate \\
>
> 17 \--cov-col Sex \\
>
> 18 \--out Height.val

![](media/image6.jpeg){width="0.3541666666666667in"
height="0.3593744531933508in"}

## Distribution of PRS

> Many PRS study publications include quantile plots that show an
> exponential in- crease in phenotypic value or / [Odd Ratios
> (OR)](#_bookmark0) among the top quantiles (e.g. an S-shaped quantile
> plot, e.g. Fig.2.3). This might lead us to believe that individuals
> with PRS values in the top quantiles have a distinctly diﬀerent
> genetic aetiology compared to the rest of the sample, or that there is
> epistasis/interactions caus- ing there substantially higher risk.
> However, when we plot a normally distributed variable (e.g. a PRS) as
> quantiles on the X-axis then we expect to observe this exponential
> pattern even when the X variable only has a linear eﬀect on the Y
> vari- able. This is because the top (and bottom) quantiles are further
> away from each other on the absolute scale of the variable and so the
> diﬀerences in their eﬀects are larger than between quantiles in the
> middle of the distribution.
>
> To understand this more, we will perform a simple simulation using
> **R**
>
> 1 R
>
> 2 \# First, we define some simulation parameters
>
> 3 n.sample \<- 10000
>
> 4 PRS.r2 \<- 0.01
>
> 5 \# Then, we simulate PRS that follow a random normal distribution
>
> Figure 2.2: An example of density plot for [PRS](#_bookmark0)

![](media/image8.png){width="1.7133333333333334in"
height="1.7124989063867018in"}

> Figure 2.3: An example of a S-shaped quantile plot

![](media/image9.jpeg){width="2.1605205599300086in"
height="2.158333333333333in"}

> 6 prs \<- rnorm(n.sample)
>
> 7 \# We can then simulate the phenotype using the following script
>
> 8 pheno \<- prs + rnorm(n.sample,mean=0,
> sd=sqrt(var(prs)∗(1-PRS.r2)/(PRS.r2)))
>
> 9 \# We can examine the relationship between the phenotype and prs
>
> 10 \# using linear regression
>
> 11 summary(lm(pheno\~prs))
>
> 12 \# Which shows that we have the expected PRS R2
>
> 13 \# Group the phenotype and PRS into a data.frame
>
> 14 info \<- data.frame(SampleID=1:n.sample, PRS=prs, Phenotype=pheno)
>
> 15 \# Then we can generate the quantile plot.
>
> 16 \# To save time, we will load in the quantile plot script from
> Software
>
> 17 source(\"./Software/Quantile.R\")
>
> 18 \# Then we can plot the quantile plot using quantile_plot option
>
> 19 quantile_plot(info, \"Height\", 100)

![](media/image6.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}

## Gene Set Analysis

> Currently, most [PRS](#_bookmark0) analyses have been performed on a
> genome-wide scale, disre- garding the underlying biological pathways.
> Udler et al. 2018 suggest that grouping [Single Nucleotide
> Polymorphisms (SNPs)](#_bookmark0) into biological functional groups
> can lead to [PRS](#_bookmark0) that are more relevant to clinical
> risk.
>
> In this practical, we will go through some common file formats for
> gene-set analysis and will then calculate some gene-set (or pathway)
> based [PRS](#_bookmark0).

### Molecular Signatures Database (MSigDB)

> The [Molecular Signatures Database (MSigDB)](#_bookmark0) oﬀers an
> excellent source of gene-sets, including the hallmark genes, gene-sets
> of diﬀerent biological processes, gene-sets of diﬀerent oncogenic
> signatures etc. All gene-sets from [MSigDB](#_bookmark0) follows the
> [Gene](#_bookmark0) [Matrix Transposed file format
> (GMT)](#_bookmark0), which consists of one line per gene-set, each
> containing at least 3 column of data:
>
> 1 \[Set A\] \[Description\] \[Gene 1\] \[Gene 2\] \...
>
> 2 \[Set B\] \[Description\] \[Gene 1\] \[Gene 2\] \...
>
> Have a look at the **Reference/Sets.gmt** file.

![](media/image6.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}![](media/image2.jpeg){width="0.3229155730533683in"
height="0.3229166666666667in"}

> As [GMT](#_bookmark0) format does not contain the chromosomal location
> for each individual gene, an additional file is required to provide
> the chromosoaml location such that [SNPs](#_bookmark0) can be map to
> genes.

### General Transfer Format file

> The [General Transfer Format (GTF)](#_bookmark0) file contains the
> chromosomal coordinates for each gene. It is a **tab** separated file
> and all but the final field in each feature line must contain a value.
> \"Empty\" columns should be denoted with a '.'. You can read the full
> format specification
> [here.](http://www.ensembl.org/info/website/upload/gff.html)
>
> One column that might be of particular interest is column 3:
> **feature**, which indi- cates what feature that line of
> [GTF](#_bookmark0) represents. This allows us to select or ignore
> features that are of interest.
>
> You can find the description of each feature
> [here](http://www.sequenceontology.org/browser/obob.cgi).

### Browser Extensible Data (BED)

> [Browser Extensible Data (BED)](#_bookmark0) file (diﬀerent to the
> binary ped file from PLINK) is a file format to define genetic
> regions. It contains 3 required fields per line (**chromosome, start
> coordinate and end coordinate**) together with 9 addi- tional optional
> field.
>
> A *special* property of [BED](#_bookmark0) is that it is a 0-based
> format, i.e. chromosome starts at 0, as opposed to the usual 1-based
> format such as the PLINK format. For example, a [SNP](#_bookmark0) on
> chr1:10000 will be represented as
>
> 1 1 9999 10000
>
> ![](media/image6.jpeg){width="0.3541655730533683in"
> height="0.3593744531933508in"}

### Gene-Set Enrichment Analysis

> Now we have gone through all the files involved in gene-set analysis,
> we should consider one of the most important aspects of gene-set (or
> pathway) enrichment analyses, which is the diﬀerent types of testing
> that we can perform when doing them:

#### Self-Contained vs Competitive Testing {#self-contained-vs-competitive-testing .unnumbered}

> The null-hypothesis of self-contained and competitive test statistics
> is diﬀerent:

-   **Self-Contained** - None of the genes within the gene-set are
    > associated with the phenotype

-   **Competitive** - Genes within the gene-set are no more associated
    > with the phenotype than genes outside the gene-set

> Therefore, a bigger gene-set will have a higher likelihood of having a
> significant
>
> *P* -value from self-contained test, which is not desirable.

## Gene-Set Based PRS Analysis

> Having learnt about the basics of gene-set analyses, we are now ready
> to perform gene-set association analyses using PRSet.
>
> To perform the PRSet analysis and obtain the set based PRS and
> competitive *P* -value, simply provide the [GTF](#_bookmark0) file and
> the [GMT](#_bookmark0) file to PRSice and specify the number of
> permutation for competitive P-value calculation using the \--set-perm
> option
>
> 1 Rscript Software/PRSice.R \\
>
> 2 \--prsice Software/PRSice_mac \\
>
> 3 \--base Base_Data/GIANT_Height.txt \\
>
> 4 \--target Target_Data/TAR \\
>
> 5 \--A1 Allele1 \\
>
> 6 \--A2 Allele2 \\
>
> 7 \--snp MarkerName \\
>
> 8 \--pvalue p \\
>
> 9 \--stat b \\
>
> Figure 2.4: An example of the multi-set plot. Sets are sorted based on
> their self- contained *R*^2^. *Base* is the genome wide
> [PRS](#_bookmark0)

![](media/image10.jpeg){width="3.9582283464566927in"
height="4.009687226596675in"}

> 10 \--beta \\
>
> 11 \--binary-target F \\
>
> 12 \--pheno Target_Data/TAR.height \\
>
> 13 \--cov Target_Data/TAR.covariate \\
>
> 14 \--out Height.set \\
>
> 15 \--gtf Reference/Homo_sapiens.GRCh38.86.gtf \\
>
> 16 \--wind-5 5kb \\
>
> 17 \--wind-3 1kb \\
>
> 18 \--msigdb Reference/Sets.gmt \\
>
> 19 \--multi-plot 10 \\
>
> 20 \--set-perm 1000

![](media/image5.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}![](media/image5.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}

![](media/image6.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}
